even_fib_nums.js

```diff
@@ -5,22 +5,47 @@
  * @param  {Number} maxFibValue
  * @return {Number} sum
  */
-function _sumFibs( maxFibValue ) {
-  var sum = 0;
+function _sumFibs(maxFibValue) {
+  var sum = 2;
+  let fibArray = [1, 2];
 
-  // do your work here
+  if (maxFibValue > 2) {
+    while (fibArray[fibArray.length - 1] <= maxFibValue) {
+      let length = fibArray.length;
+      let current = fibArray[length - 2] + fibArray[length - 1];
+      fibArray.push(current);
+
+      if ((current % 2 === 0) && (current <= maxFibValue)) {
+        sum += current;
+      }
+    }
+
+  } else {
+    return maxFibValue;
+  }
 
   return sum;
 }
 
+
 // bonus round
-function _highestFibonacciNumber (maxFibValue){
+function _highestFibonacciNumber(maxFibValue) {
   var highest = 0;
+  let fibArray = [1, 2];
+
+  if (maxFibValue > 2) {
+    do {
+
+      let length = fibArray.length;
+      fibArray.push(fibArray[length - 2] + fibArray[length - 1]);
 
-  //define your base case, validate your input
+    } while (fibArray[fibArray.length - 1] < maxFibValue);
 
+    highest = fibArray.filter(x => x <= maxFibValue).pop();
```

Comment:
So this will go through the entire array of fibonacci numbers just to find the highest one less than maxFibValue. But you could easily keep track of that in your do-while loop as well.